x-pack/plugins/session_view/public/components/ProcessTreeNode/index.tsx

```diff
@@ -0,0 +1,313 @@
+
+  const processDetails = useMemo(() => {
+    return process.getDetails();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [process.events.length]);
+
+  const hasExec = useMemo(() => {
+    return process.hasExec();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [process.events.length]);
+
+  const alerts = useMemo(() => {
+    return process.getAlerts();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [process.events.length]);
+
+  const styles = useStyles({ depth, hasAlerts: !!alerts.length });
+
+  useLayoutEffect(() => {
+    if (searchMatched !== null && textRef.current) {
+      const regex = new RegExp(searchMatched);
+      const text = textRef.current.textContent;
+
+      if (text) {
+        const html = text.replace(regex, (match) => {
+          return `<span data-test-subj="processNodeSearchHighlight" style="${styles.searchHighlight}">${match}</span>`;
+        });
+
+        // eslint-disable-next-line no-unsanitized/property
+        textRef.current.innerHTML = html;
+      }
+    }
+  }, [searchMatched, styles.searchHighlight]);
+
+  if (!processDetails) {
+    return null;
+  }
+
+  const { tty } = processDetails.process;
+
+  const renderChildren = () => {
+    const children = process.getChildren(showGroupLeadersOnly);
+
+    if (!childrenExpanded || !children || children.length === 0) {
+      return null;
+    }
+
+    const newDepth = depth + 1;
+
+    return (
+      <div css={styles.children}>
+        {children.map((child: Process) => {
+          return (
+            <ProcessTreeNode
+              key={child.id}
+              process={child}
+              depth={newDepth}
+              onProcessSelected={onProcessSelected}
+            />
+          );
+        })}
+      </div>
+    );
+  };
+
+  const getExpandedIcon = (expanded: boolean) => {
+    return expanded ? 'arrowUp' : 'arrowDown';
+  };
+
+  const renderButtons = () => {
+    const buttons = [];
+    const childCount = process.getChildren().length;
+
+    if (isSessionLeader) {
+      const groupLeaderCount = process.getChildren(true).length;
+      const sameGroupCount = childCount - groupLeaderCount;
+
+      if (sameGroupCount > 0) {
+        buttons.push(
+          <EuiToolTip
+            key="samePgidTooltip"
+            position="top"
+            content={
+              <p>
+                <FormattedMessage
+                  id="xpack.sessionView.groupLeaderTooltip"
+                  defaultMessage="Hide or show other processes in the same 'process group' (pgid) as the session leader. This typically includes forks from bash builtins, auto completions and other shell startup activity."
+                />
+              </p>
+            }
+          >
+            <EuiButton
+              key="child-processes-button"
+              css={styles.getButtonStyle(ButtonType.children)}
+              onClick={() => setShowGroupLeadersOnly(!showGroupLeadersOnly)}
```

Comment:
Nit: New function reference will be created on every render. Same for the other anonymous functions passed as props below. Maybe move out to the top. This could be `const onChildProcessButtonClick = () => setShowGroupLeadersOnly(!showGroupLeadersOnly);`